src/note.c

```diff
@@ -5,19 +5,37 @@
 
 // --------
 
-void Note_init_at(Note* note, int8_t pitch, int8_t length) {
+static float get_absolute_granularity(const Note* note);
+
+static size_t absolute_length_to_samples(float length);
```

Comment:
Might want to move that (and related functions?) into the `note_conversion` module (or drop that altogether).

Reply:
Can be done in a later PR; haven't decided yet what the ideal solution would be.